src/dotnet-roslyn-tools/PRFinder/Hosts/GitHub.cs

```diff
@@ -12,15 +16,38 @@ public class GitHub : IRepositoryHost
     private static readonly Regex IsGitHubReleaseFlowCommit = new(@"^Merge pull request #\d+ from dotnet/merges/");
     private static readonly Regex IsGitHubMergePRCommit = new(@"^Merge pull request #(\d+) from");
     private static readonly Regex IsGitHubSquashedPRCommit = new(@"\(#(\d+)\)(?:\n|$)");
+    private readonly string _repoUrl;
 
-    public string GetCommitUrl(string repoUrl, string commitSha)
-        => $"{repoUrl}/commit/{commitSha}";
+    private readonly HttpClient _httpClient;
+    private readonly ILogger _logger;
 
-    public string GetDiffUrl(string repoUrl, string previousSha, string currentSha)
-        => $"{repoUrl}/compare/{previousSha}...{currentSha}?w=1";
+    public GitHub(string repoUrl, ILogger logger)
+    {
+        _repoUrl = repoUrl;
+        _logger = logger;
+        _logger.LogTrace($"Creating GitHub repository host with base url: {repoUrl}");
+
+        var split = _repoUrl.Split('/', StringSplitOptions.RemoveEmptyEntries);
+        var owner = split[^2];
+        var repo = split[^1];
+        _httpClient = new HttpClient(new LoggingHandler(logger))
+        {
+            BaseAddress = new Uri($"https://api.github.com/repos/{owner}/{repo}/pulls/"),
+        };
+
+        _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/vnd.github.text+json"));
```

Comment:
Might be OK if we don't hit the rate limit issue in GitHub.
We can wait & test if anything will be wrong.